arch/riscv/core/smp.c

```diff
@@ -74,7 +74,11 @@ void arch_secondary_cpu_init(int hartid)
 #endif
 #ifdef CONFIG_SMP
 	irq_enable(RISCV_IRQ_MSOFT);
-#endif
```

Comment:
I think technically this could stay here, since plic affinity depends on SMP. 

This is still correct as well, but we avoid nesting ifdefs